tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_friendship.c

```diff
@@ -361,6 +361,20 @@ static void test_friend_group(void)
 	PASS();
 }
 
+/* LPN disable friendship dummy
+ *
+ * Dummy friendship node for test_lpn_disable
+ */
+static void test_friend_lpn_disable_dmy(void)
+{
+	/* Wait for LPN node to toggle lpn_set */
+	k_sleep(K_SECONDS(1));
```

Comment:
No need for this, I believe? We should just be waiting for LPN to send the friend request, which we should be able to receive right away, right?

Reply:
I tried this at first, but the simulation environment crashed for the friendship node without the delay for some reason (Before adding the fix to the LPN module):

`p_2G4: WARNING: (src/p2G4_com.c:110): Low level communication with device 1 broken during Abort reevaluation (tried to get 16 got 0 bytes) (most likely the device was terminated)`
`p_2G4: ERROR: (src/p2G4_com.c:112): Exiting`
`timeout: the monitored command dumped core`
`/home/anderss/repos/ncs/zephyr/tests/bluetooth/bsim_bt/bsim_test_mesh/tests_scripts/friendship/../../_mesh_test.sh: line 72:  7128 Segmentation fault      timeout 20 $@`

Hence the workaround. This will of course make the test fail, which is kind of what we want it to do, but the cause will be quite unclear to the user. If you have some notion of how to fix this without the delay I'm all for it, would be much cleaner that way.

Reply:
Also uncertain if this behaviour might indicate another potential bug in the friendship module

Reply:
Could you check again? It seems to be working fine at my desk with your latest changes and without this `k_sleep`.

Reply:
Might be alleviated by the evt_wait below, perhaps?

Reply:
> Might be alleviated by the evt_wait below, perhaps?

Yeah, seems so.

Reply:
It works now 👍

Reply:
Well I spoke too soon. @PavelVPV could you try to run the test while commenting out the fixes in lpn.c? I belive that i just tested it the second time with the fixes in place, which as described above will not trigger the error.

Reply:
Yes, it crashes because lpn tries to process friend offer here after being disabled as the lpn state get overwritten. Backtrace:
> Thread 9 "bs_nrf52_bsim_t" received signal SIGSEGV, Segmentation fault.                                                                           
[Switching to Thread 0xf43fdb40 (LWP 55499)]                           
0x08103e73 in bt_mesh_lpn_friend_offer (rx=0xf43fcf64, buf=0xf43fcf58)   
    at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/lpn.c:588
588                     if (!lpn->sub->keys[i].valid) {
(gdb) bt
#0  0x08103e73 in bt_mesh_lpn_friend_offer (rx=0xf43fcf64, buf=0xf43fcf58)
    at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/lpn.c:588
#1  0x080c7ae3 in ctl_recv (rx=0xf43fcf64, hdr=4 '\004', buf=0xf43fcf58, seq_auth=0xf43fcee0)
    at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/transport.c:923
#2  0x080c8575 in trans_unseg (buf=0xf43fcf58, rx=0xf43fcf64, seq_auth=0xf43fcee0)
    at /home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/transport.c:972
#3  0x080cf13f in bt_mesh_trans_recv (buf=0xf43fcf58, rx=0xf43fcf64)

state:
> (gdb) p lpn->state
$4 = BT_MESH_LPN_WAIT_OFFER

Reply:
Ok, seems legit. The question is how we deal with it in the test. After the applied fixes we should hopefully not experience this situation anymore, but that is hardly the point. The point is if it acceptable for the test to fail in this manner if a simmilar  bug re-appears. In other words if its ok that the test fails in a "strange" manner. Im not quite sure, so I leave it to you guys to decide if we want to put in the old fix again, or if it is fine as is.

Reply:
Should assertion be added if it causes a segmentation fault?

Reply:
I have discussed this with @PavelVPV, and we have agreed that the test is good as it is.

Reply:
Yep, but I meant lpn.c not test

Reply:
> Should assertion be added if it causes a segmentation fault?

I'm not sure really